src/libsystemd/sd-varlink/sd-varlink.c

```diff
@@ -1066,18 +1066,18 @@ static int varlink_sanitize_parameters(sd_json_variant **v) {
 
         assert(v);
 
-        /* Varlink always wants a parameters list, hence make one if the caller doesn't want any */
-        if (!*v)
-                return sd_json_variant_new_object(v, NULL, 0);
-        if (sd_json_variant_is_null(*v)) {
-                sd_json_variant *empty;
-
-                r = sd_json_variant_new_object(&empty, NULL, 0);
+        /* Convert NULL or JSON null to empty object for method handlers */
+        if (!*v) {
+                r = sd_json_variant_new_object(v, NULL, 0);
                 if (r < 0)
                         return r;
-
+                return 0;
+        }
+        if (sd_json_variant_is_null(*v)) {
                 sd_json_variant_unref(*v);
-                *v = empty;
+                r = sd_json_variant_new_object(v, NULL, 0);
+                if (r < 0)
+                        return r;
                 return 0;
         }
```

Comment:
you can merge these two if expressions into one. note that sd_json_variant_unref() happily accepts NULL as input in which case it is a NOP.